app/src/main/java/org/mozilla/focus/widget/Cursor.kt

```diff
@@ -72,8 +69,7 @@ class Cursor(context: Context, attrs: AttributeSet) : View(context, attrs) {
     override fun onLayout(changed: Boolean, left: Int, top: Int, right: Int, bottom: Int) {
         super.onLayout(changed, left, top, right, bottom)
 
-        if (changed && isInit) {
-            isInit = false
```

Comment:
Is it okay to remove this isInit? Do we need it to keep track of whether the cursor has been properly initialized by the calling code (setting maxBounds, etc), even if the calling fragment has been destroyed?

Reply:
fwiw, I think `isInit` was not being used properly – I think `isInit` was added because `changed` was not being used in the original code and we wanted to prevent the cursor position from getting set more than once since onLayout may get called several times when the view is being created. Therefore, I think we can remove isInit here provided we check `changed`.

That being said, I moved `isInitialPositionSet` to CursorViewModel, to prevent the Cursor from jumping around after the position is initially set and you do point out a bug: if the CursorView is removed but the CursorViewModel is not, `isInitialPositionSet` will remain set and we won't reposition the cursor to the center of the screen when the new view is created (which may or may not be desired...).

I guess this is why the viewModel ideally shouldn't have a dependency on the view, huh? :)

I'll think about this for the next revision.